File: java/Loops.java
import java.util.Random;

public class Loops {

  public static void main(String[] args) {
    final int iterations = Integer.parseInt(args[0]);
    simulate(iterations);
  }

  private static void simulate(int iterations) {
    long start = System.currentTimeMillis();
    final int SAMPLE_SIZE = 23;
    Random rand = new Random();
    rand.setSeed(start);
    int duplicates = 0;
      for (long x=0; x < iterations; x++) {
      int data[] = new int[365];
      for (int i=0; i < SAMPLE_SIZE; i++) {
        int number = rand.nextInt(365);
        if (data[number] == 1) {
          duplicates++;
          break;
        } else {
          data[number] = 1;
        }
      }
    }
    System.out.println("iterations: " + iterations);
    System.out.println("sample-size: " + SAMPLE_SIZE);
    double results = duplicates * 100.0 / iterations;
    System.out.format("percent: %.2f%n", results);
    long end = System.currentTimeMillis();
    double diff = (end-start) / 1_000.0;
    System.out.format("seconds: %.3f%n", diff);
  }

}

File: kotlin/loops.kt
import java.util.Random

fun main(args: Array<String>) {
  simulate()
}

fun simulate() {
  val start = System.currentTimeMillis();
  val iterations = 2000 * 1000
  val sample_size = 23
  val random = Random()

  var count = 0
  for (i in 1..iterations) {
    val data = IntArray(365);
    for (n in 1..sample_size) {
      val number = random.nextInt(365)
      if (data[number] == 1) {
        count++
        break
      } else {
        data[number] = 1
      }
    }
  }
  println("iterations: ${iterations}")
  println("sample-size: ${sample_size}")
  val percent = count * 100.0 / iterations
  println("percent: %.2f".format(percent))
  val end = System.currentTimeMillis();
  val diff = (end - start) / 1000.0
  println("seconds: ${diff}")
}
